tests/ui/invalid-compile-flags/crate-type-flag.rs

```diff
@@ -17,12 +17,11 @@
 //@[staticlib] compile-flags: --crate-type=staticlib
 //@[staticlib] check-pass
 
-//@[dylib] ignore-musl (dylibs are not supported)
-//@[dylib] ignore-wasm (dylibs are not supported)
+//@[dylib] needs-dynamic-linking
```

Comment:
Unfortunately this is not quite accurate, see https://github.com/rust-lang/rust/issues/132309#issue-2620222352, wasm targets (wasm base) supports dynamic linking but does not support `dylib` (it does support `cdylib`). We still need to ignore musl right now because musl libc is still currently statically linked https://github.com/rust-lang/rust/issues/132350#issuecomment-2447389856.

Reply:
> However, I think the `ignore-wasm` here is wrong and can be removed.

would it not fail if removed (given your explanation)

Reply:
No you're right, I think I added that sentence on the wrong review comment lol